deep_net.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+
+import pickle
+import random
+from time import time
+import numpy as np
+import tensorflow as tf
+import tensorflow.keras.layers as layers
+from sklearn.metrics import accuracy_score
+from snippets import get_random_parameters
+
+def train_neural_net(train_set, test_set, label, training_time, store_intermediate_results=False):
+    """ Train neural networks with randomly generated hyperparameters: learning rate, learning rate
+    decay and number of layer. Model: fully connected NN for multiclass classification.
+
+    Args:
+        train_set(pandas df):                       training set
+        test_set(pandas df):                        test set, same features as in training set
+        label(str):                                 feature with data labels
+        training_time(float):                       training time (in hours)
+        store_intermediate_results(bool, optional): True for saving model at the end of training, False otherwise
+
+    Returns:
+        all_models(list of keras.model):            neural networks models
+        accuracy(list of 2 (N,) np arrays):         in sample (accuracy[0]) and out of sample (accuracy[1]) accuracy
+        parameters(list of 3 (N,) np arrays):       hyperparameters: learning rates (parameters[0]),
+                                                                     learning rate decays (parameters[1]),
+                                                                     num fo layers (parameters[2]),
+    """
+
+    PARAM_RANGE = {
+        "learning_rate": (1e-1, 1e-6),
+        "layers": (10, 1e3),
+        "lr_decay": (0.5, 1e-8)
+    }
+    BATCH_NORM = False
+    EPOCHS = 100
+    BATCH_SIZE = 1000
+    BETA_1 = 0.9
+    BETA_2 = 0.999
+
+    PARAM_FILE = "deep_net_stats"
+    MODEL_FILE = "deep_net_model_"
+
+    all_models = []
+    accuracy = [[], []]
+    learning_rate = []
+    lr_decay = []
+    deep_layers = []
+
+    features = list(train_set)
+    features.remove(label)
+    num_class = len(np.unique(train_set[label]))
+
+    i = 0
```

Comment:
Prefer a more explicit name such as `index` when not in a for loop.
Non-explicit variable names get confusing when carried around a large block of code.